tests/unit/test_index.py

```diff
@@ -162,6 +162,17 @@ def test_index(self):
         with f:
             f.search(input_fn=random_docs(2), output_fn=get_result)
 
+    def test_index_request_generator(self):
```

Comment:
what are we really testing here?

Reply:
my bad, let me create new pull req cuz i forgot to create new branch in my fork

Reply:
you can change it in this PR if you need to change it.

But I fail to see what this test is actually doing.

Reply:
i'm checking can we get req list even if doc is starting with url. in the old code it's throwing error on this part